demo/src/search/search-page.js

```diff
@@ -35,14 +36,24 @@ const createResultsEl = (results) => {
 
 router.addRoute('search', () => {
   document.querySelector('.container').replaceChildren(...createContentEl());
-  document.querySelector('#search-bar').addEventListener('keyup', (event) => {
-    if (event.key === 'Enter') {
+
+  const spinner = new SpinnerComponent().attach('.container');
+
+  document.querySelector('#search-bar').addEventListener('keyup', async (event) => {
+    if (event.key === 'Enter' && spinner.hidden) {
```

Comment:
In the case where the user changes the query and presses <kbd>Enter</kbd> while the spinner is visible, we won't send the search with the new value, right?

Reply:
That was my intention at first. Regardless, thanks to your comment I've found a way to abort the fetching of results so this is layer of protection is no longer necessary, take a look at the refactored code.